cloud/microsoft/office365/exchange/mode/emailactivity.pm

```diff
@@ -187,36 +191,43 @@ sub manage_selection {
     
     $self->{active} = { active => 0, total => 0, report_date => '' };
     $self->{global} = { send_count => 0, receive_count => 0 , read_count => 0 };
-    $self->{mailboxes} = {};
+    $self->{users} = {};
+
+    my $results = $options{custom}->office_get_exchange_activity(param => "period='D7'");
```

Comment:
@UrBnW why do you keep an hardcoded period here?

Reply:
The various API calls require to set either [period or date](https://docs.microsoft.com/en-us/graph/api/reportroot-getemailactivityuserdetail?view=graph-rest-1.0).
Period is D7, D30, D90, or D180.
Date is YYYY-MM-DD, for which you would like to view the users who performed any activity.

So, using period `D7` in the first API call mainly allows us to get 2 things :
- the number of inactive objects ;
- the date the report has been generated, which will be used as the date parameter to the second API call.

We can't directly do the second call, as we don't know the last refresh date (which varies and is sometimes a few days ago).
And this second call does not return inactive objects information.

Reply:
This is under investigation on our side. ANy feedback about double-call overhead? Is this fast enough to satisfy everybody? Do you have a comparison of what it implies in terms of exec time.

Reply:
Yes, I run all these modified plugins since weeks now.
Their execution time varies between one and two seconds.
Seems largely acceptable, keeping in mind that it's for about 10 services overall, and that the check frequency can be hour-based, instead of (generally) minute-based, as the reported data is daily-based, not continuously refreshed.
I run them hourly, everything's fine 👍